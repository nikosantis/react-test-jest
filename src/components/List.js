import React from 'react'

export const List = ({ title, list }) => {
  return (
    <section>
      <h1 className='big'>{ title }</h1>
      <ul>
        {list.map(item => (
          <li key={item.id}>
            { item.name }
          </li>
        ))}
      </ul>
    </section>
  )
}
