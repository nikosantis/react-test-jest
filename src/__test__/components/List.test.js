import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { List } from '../../components/List'

describe('Probando componente <List />', () => {
  const fruits = [
    { name: 'Fresa', id: 1 },
    { name: 'Manzana', id: 2 },
    { name: 'Sandía', id: 3 },
    { name: 'Banana', id: 4 }
  ]
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<List title='Frutas' list={fruits} />)
  })

  it('Validar nodos', () => {

    // validar si existe nodo
    expect(wrapper.find('h1').exists()).toBe(true)

    // validar si tiene clase css
    expect(wrapper.find('h1').hasClass('big')).toBe(true)

    // validar cantidad de elementos hijos
    expect(wrapper.find('ul').children().length).toBe(4)

    // validar el contenido de texto
    expect(wrapper.find('ul').childAt(2).text()).toBe('Sandía')
    // console.log(wrapper.find('li').first().html())
    // console.log(wrapper.find('li').last().html())
    // console.log(wrapper.find('ul').childAt(2).html())

    // validar el tipo de nodo / elemento
    expect(wrapper.find('ul').childAt(2).type()).toBe('li')

    // validar el html
    // validar el contenido de texto
    expect(wrapper.find('ul').childAt(2).html()).toBe('<li>Sandía</li>')
  })

  it('Validar actualizaciones en props', () => {
    // validar cantidad de elementos li
    expect(wrapper.find('li').length).toBe(4)

    wrapper.setProps({
      list: [
        { name: 'kiwi', id: 5 }
      ]
    })

    // validar cantidad de elementos li
    expect(wrapper.find('li').length).toBe(1)

    wrapper.setProps({
      title: 'Super frutas'
    })

    // validar cantidad de elementos li
    expect(wrapper.find('.big').text()).toBe('Super frutas')
  })

  it('Validar que coincida con Spapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})