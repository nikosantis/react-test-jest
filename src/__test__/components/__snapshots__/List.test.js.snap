// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <List /> Validar que coincida con Spapshot 1`] = `
<section>
  <h1
    className="big"
  >
    Frutas
  </h1>
  <ul>
    <li
      key="1"
    >
      Fresa
    </li>
    <li
      key="2"
    >
      Manzana
    </li>
    <li
      key="3"
    >
      Sand√≠a
    </li>
    <li
      key="4"
    >
      Banana
    </li>
  </ul>
</section>
`;
